### PR TITLE
Catch bcftools concat errors and atomize VCF parsing processes

### PR DESCRIPTION
**Pull request by groodri**
To make it easier to test, especially bcftools behavior around `bcftools concat` and `bcftools sort`

### CODE REVIEW

**Comment by groodri** [comment]
Still doesn't catch bcftools error messages to throw exit code 1

File: subworkflows/modules/bin.nf [hunk 1]
```diff
@@ -111,8 +113,9 @@ process FilterValidateVariants{
         --max_sap 0 \
         --min_rel_ratio $params.snp_filters.min_rel_ratio \
         --min_abq $params.snp_filters.min_abq
+        """
 
-        vcf_filter.py -i $indel_vcf -o ${indel_vcf.simpleName}.filtered.indel.vcf \
+        vcf_filter.py -i $indel_vcf -o ${indel_vcf.simpleName}_filtered.indel.vcf \
```
Comment (da-i) [inline subworkflows/modules/bin.nf:118]: Can you make the naming consistent? e.g. all underscores?

File: subworkflows/modules/bin.nf [hunk 2]
```diff
@@ -94,14 +94,16 @@ process FilterValidateVariants{
     label 'many_low_cpu_high_mem'
 
     input:
-        tuple path(snp_vcf), path(indel_vcf), val(X), path(perbase_table)
+        tuple val(X), path(vcf), path(perbase_table)
+        val(mode)
 
     output:
-        tuple path("${snp_vcf.simpleName}.filtered.snp.vcf"), path("${indel_vcf.simpleName}.filtered.indel.vcf")
+        tuple path("${snp_vcf.simpleName}_filtered.snp.vcf"), path("${indel_vcf.simpleName}_filtered.indel.vcf")
```
Comment (da-i) [inline subworkflows/modules/bin.nf:101]: idem

**Review by da-i — COMMENTED** [review]
Please make filenames consistent

**Comment by groodri** [comment]
> Please make filenames consistent

This is an artefact from merging two commits into one. I've fixed it but need to test again to be sure no other artefacts remain.

**Review by da-i — APPROVED** [review]

**Pull request merged**